### ISSUE TITLE
VSIX Files Adding To /tmp Each Hour VSCode is Open.

### ISSUE THREAD

**Issue by vindicatorr** [microsoft/vscode-cpptools#3923]
Type: General
**Describe the bug**
- OS and Version: Linux x64 5.1.15.a-1-hardened
- VS Code Version: Code - Insiders 1.36.0-insider (c089daa, 2019-06-07T05:15:26.964Z)
- C/C++ Extension Version: 0.23.1
- Other extensions you installed (and if the issue persists after disabling them): Unrelated
- A clear and concise description of what the bug is. https://github.com/microsoft/vscode/issues/76911
Everything has already been mentioned there. The short of it is that once a project is opened that uses this extension, a vsix file is created in /tmp every 1 hour, and the file is NEVER deleted.

**To Reproduce**
1. Open any project that uses this extension.
2. Monitor the /tmp folder for created vsix files over time.
3. Close VSCode
4. Note /tmp still has replicating vsix files.

**Expected behavior**
Don't automatically download vsix files when update is set to manual.
Clean up vsix files.

**Screenshots**
Not needed.

**Additional context**
I set the extension log verbosity to Debug and saw nothing related to vsix updates.
Make sure to look at the linked issue at vscode for details. We can continue from here if it is related to this extension and not a vscode issue.
Log (Main):
```
[2019-07-12 16:38:24.212] [main] [info] update#setState idle
[2019-07-12 16:38:24.213] [main] [info] update#ctor - manual checks only; automatic updates are disabled by user preference
```
$ ls -lat /tmp/*.vsix
-rw------- 1 \<uid> \<gid> 19283261 Jul 12 17:42 /tmp/tmp-10323wGcOCsJMxKxG.vsix
-rw------- 1 \<uid> \<gid> 19283261 Jul 12 16:41 /tmp/tmp-10323LYATa11ywpfA.vsix

**Comment by sean-mcmanus** [comment]
You can set your C_Cpp.updateChannel Setting to "Default" as a workaround (or upgrade to 0.24.0). We have another workitem to check the extensions.autoUpdate. The "Insider" update channel is causing us to download the 0.24.0 vsix -- the mechanism was intended to only apply to "insider" vsix's and not released ones.

Do you know why the downloaded vsix is not installing? Repeatedly downloading seems buggy, as is the not deleting of tmp files.

**Comment by vindicatorr** [comment]
Hmmm, yeah. I changed updateChannel to Default.
I'll switch it back to Insiders after the extension follows the autoUpdate rule.

I have no idea why it isn't installing. I never wanted it to download OR install.
```
...
    "extensions.autoUpdate": false,
    "update.mode": "manual",
    "update.enableWindowsBackgroundUpdates": false,
    "C_Cpp.updateChannel": "Default"
```
I only wanted to be notified of updates so I can see what's in the changelog and install it IF I want to.
When I do manually update, it has no problems (or at least I don't notice any).

I just confirmed that there are NO vsix files in /tmp after changing to Default. I gave it enough time so if it was going to download, it would have by now.
I found it enlightening to learn that vscode itself doesn't handle extension updates but rather the extensions themselves.
In other words I would've expected vscode to be a master that controls the main update/install process whose sole purpose is to hand off the process to the extension. Apparently it would seem that isn't the case.
The process would have like maybe 4 possible functions:
1) Check for updates
2) Download
3) Install
4) Remove

Every extension would implement each of these stubs and vscode would call on them.
The only guaranteed call would be to Check for updates, though I suppose a setting could be made to let that be a manual process as well.

**Comment by sean-mcmanus** [comment]
VS Code normally does handle the updates, but it doesn't allow for an Insiders mechanism, so we wrote an auto-update to Insiders mechanism ourselves. However, there's no notification to opt-in to a particular insider release once you're on the Insider channel.

**Comment by plwalsh** [comment]
Any movement on this one recently? My RHEL machine periodically locks up when the `/tmp` directory becomes full due to this issue.

**Comment by sean-mcmanus** [comment]
No change. You can probably workaround this via disabling the extensions.autoUpdate or setting C_Cpp.updateChannel to "Default".

**Comment by bobbrow** [comment]
We need to investigate this.  I see it on my Ubuntu installation as well.

**Comment by sean-mcmanus** [comment]
Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/0.28.0 .
